subprojects/core/src/test/groovy/org/gradle/initialization/DefaultSettingsLoaderTest.groovy

```diff
@@ -15,61 +15,124 @@
+        //noinspection GroovyAssignabilityCheck
+        process(mockGradle, mockBuildLayout, mockClassLoaderScope, _) >> { gradle, settingsLocation, clasLoaderScope, startParameter ->
+            def mockResultSettingsScript = Mock(ScriptSource) {
+                getDisplayName() >> "foo"
+            }
 
-    def gradle = Mock(GradleInternal)
-    def settings = Mock(SettingsInternal)
-    def state = Mock(SettingsState)
-    def buildLayout = new BuildLayout(null, FileUtils.canonicalize(new File("someDir")), null, Stub(ScriptFileResolver))
-    def buildLayoutFactory = Mock(BuildLayoutFactory)
-    def settingsScript = Mock(ScriptSource)
-    def startParameter = new StartParameterInternal()
-    def classLoaderScope = Mock(ClassLoaderScope)
-    def settingsProcessor = Mock(SettingsProcessor)
-    def buildInCommand = Mock(BuiltInCommand)
-    def settingsHandler = new DefaultSettingsLoader(settingsProcessor, buildLayoutFactory, [buildInCommand])
-
-    void findAndLoadSettingsWithExistingSettings() {
-        when:
-        def projectRegistry = Mock(ProjectRegistry)
-        def projectDescriptor = Mock(DefaultProjectDescriptor) {
-            getPath() >> ":"
+            def mockResultSettings = Mock(SettingsInternal) {
+                getProjectRegistry() >> mockProjectRegistry
+                getSettingsScript() >> mockResultSettingsScript
+                getStartParameter() >> startParameter
+            }
+
+            def mockResultState = Mock(SettingsState) {
+                getSettings() >> mockResultSettings
+            }
+            return mockResultState
         }
-        def services = Mock(ServiceRegistry)
-        startParameter.setCurrentDir(buildLayout.settingsDir)
-
-        settings.getProjectRegistry() >> projectRegistry
-        projectRegistry.getAllProjects() >> Collections.singleton(projectDescriptor)
-        projectDescriptor.getProjectDir() >> buildLayout.settingsDir
-        projectDescriptor.getBuildFile() >> new File(buildLayout.settingsDir, "build.gradle")
-        gradle.getStartParameter() >> startParameter
-        gradle.getServices() >> services
-        gradle.getIdentityPath() >> Path.ROOT
-        buildLayoutFactory.getLayoutFor(_) >> buildLayout
-        gradle.getClassLoaderScope() >> classLoaderScope
-        1 * settingsProcessor.process(gradle, buildLayout, classLoaderScope, startParameter) >> state
-        _ * state.settings >> settings
-        1 * settings.settingsScript >> settingsScript
-        1 * settingsScript.displayName >> "foo"
+    }
+
+    void setup() {
+        startParameterInternal.setCurrentDir( projectRootDir )
+    }
+
+    private logger = new ToStringLogger()
+    private builtInCommands = [new InitBuiltInCommand(), new HelpBuiltInCommand()]
+    private DefaultSettingsLoader settingsLoader = new DefaultSettingsLoader(mockSettingsProcessor, mockBuildLayoutFactory, builtInCommands, logger)
+
+    def "running default task loads settings"() {
```

Comment:
💭 This test is a "re-imagining" of the previous only test in this class, which ensured that the state retrieved from the settings loader was the only the processor was given.  Now, we dynamically create a new state each time and check that it isn't null.  Testing the logging ensures the same methods ran - the previous test for instance equality of state doesn't seem too valuable to be worth perfectly preserving here.